src/createApp.ts

```diff
@@ -512,6 +513,20 @@ function addIdentifier(app: App, id: Identifier) {
 	};
 }
 
+function createCustomWidget(app: App, id: string) {
+	const customFactories = customElementFactories.get(app);
+
+	return app.defaultWidgetStore.get(id).then((state: any) => {
```

Comment:
It seems like we are sort of expecting `state.type` to be there and relying on chained promise rejections to determine that, including calling `customFactory()` when it could very well be `undefined`.  While we haven't converted app yet to `strictNullChecks` I think we should avoid adding code that we know we would need to rewrite later on.